test/policies/adopter_application_policy_test.rb

```diff
@@ -0,0 +1,189 @@
+require "test_helper"
+
+# See https://actionpolicy.evilmartians.io/#/testing?id=testing-policies
+class AdopterApplicationPolicyTest < ActiveSupport::TestCase
+  include PetRescue::PolicyAssertions
+
+  context "#index?" do
+    setup do
+      @policy = -> {
+        AdopterApplicationPolicy.new(AdopterApplication, user: @user)
+      }
+      @action = -> { @policy.call.apply(:index?) }
+    end
+
+    context "when user is nil" do
+      setup do
+        @user = nil
+      end
+
+      should "return false" do
+        assert_equal @action.call, false
+      end
+    end
+
+    context "when user is adopter without adopter account" do
+      setup do
+        @user = create(:user)
+      end
+
+      should "return false" do
+        assert_equal @action.call, false
+      end
+    end
+
+    context "when user is adopter without profile" do
+      setup do
+        @user = create(:user, :adopter_without_profile)
+      end
+
+      should "return false" do
+        assert_equal @action.call, false
+      end
+    end
+
+    context "when user is adopter with profile" do
+      setup do
+        @user = create(:user, :adopter_with_profile)
+      end
+
+      should "return true" do
+        assert_equal @action.call, true
+      end
+    end
+  end
+
+  context "#create?" do
+    setup do
+      @pet = create(:pet)
+      @policy = -> {
+        AdopterApplicationPolicy.new(AdopterApplication, user: @user,
+          pet: @pet)
+      }
+      @action = -> { @policy.call.apply(:create?) }
+    end
+
+    context "when user is nil" do
+      setup do
+        @user = nil
+      end
+
+      should "return false" do
+        assert_equal @action.call, false
+      end
+    end
+
+    context "when user is adopter without adopter account" do
```

Comment:
I woud rename this when user not associated with adopter account - minor detail really, but it's the association that makes them an adopter.

Reply:
Good catch!

I do have a question for best practice in this project. Would it be better for me to:
1. Rebase this commit with the fix
2. Rebase a new commit after this commit to fix
3. Add a new commit at the end of the branch to fix

Reply:
I think it is OK to add a new commit at the end to keep it easier for you. My approach here in review could have been better, too. I started going through commits just because I wanted to see things in chunks. But, then I just looked at the whole picture via Files Changed in the end. So, I realise some of my comments on things were really no longer relevant due to later changed. Apologies!

Reply:
ca924d545f7d063ee5adeeda43cb272826fdb30f resolves